core/validatorapi/validatorapi.go

```diff
@@ -357,6 +360,66 @@ func (c Component) SubmitBeaconBlock(ctx context.Context, block *spec.VersionedS
 	return nil
 }
 
+// SubmitVoluntaryExit receives the partially signed voluntary exit.
+func (c Component) SubmitVoluntaryExit(ctx context.Context, ve *eth2p0.SignedVoluntaryExit) error {
+	vs, err := c.Validators(ctx, "head", []eth2p0.ValidatorIndex{ve.Message.ValidatorIndex})
+	if err != nil {
+		return err
+	}
+	if len(vs) != 1 {
+		return errors.New("cannot find validator", z.U64("ValidatorIndex", uint64(ve.Message.ValidatorIndex)))
+	}
+
+	pubKeyBytes, err := vs[ve.Message.ValidatorIndex].PubKey(ctx)
+	if err != nil {
+		return err
+	}
+
+	pubKey, err := core.PubKeyFromBytes(pubKeyBytes[:])
+	if err != nil {
+		return err
+	}
+
+	sigRoot, err := ve.Message.HashTreeRoot()
+	if err != nil {
+		return err
+	}
+
+	err = c.verifyParSig(ctx, core.DutyVoluntaryExit, ve.Message.Epoch, pubKey, sigRoot, ve.Signature)
+	if err != nil {
+		return err
+	}
+
+	data, err := json.Marshal(ve)
+	if err != nil {
+		return nil
+	}
+
+	parSigDate := core.ParSignedData{
+		Data:      data,
+		Signature: core.SigFromETH2(ve.Signature),
+		ShareIdx:  c.shareIdx,
+	}
+
+	parsigSet := core.ParSignedDataSet{
+		pubKey: parSigDate,
+	}
+
+	duty := core.Duty{
+		Type: core.DutyVoluntaryExit,
+		Slot: math.MaxInt64,
+	}
```

Comment:
I would use slot 0, since a zero value in golang often means "unset" and slot 0 isn't a valid eth2 slot.

Reply:
or actually, we should probably convert the epoch into a slot (1 slot in the epoch) and use that. Just incase multiple exits are requested with different epochs.

Reply:
ok, I do it but I will put a reference that it was your idea

Reply:
Done